### PR TITLE
feat(web-barbies): implement state management for blogs page

### PR DESCRIPTION
**Pull request by nawrazi**
Previously, the Blogs and Single Blog pages fetched their data directly from a json file in the public folder. In this PR, I have created api routes and slices and changed the fetching logic that way. I have also added loading shimmer animations for the blog cards.
This PR assumes that that a dedicated error page will be made in the future, and currently just shows a plain text message during errors.

The loading shimmer component for blogs list:
Screenshot 2023-05-12 at 11 10 28 AM

The loading shimmer component for related blogs:
Screenshot 2023-05-12 at 11 09 37 AM

### CODE REVIEW

File: Web/starter_project_barbies/pages/api/blogs/index.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+
+interface Blog {
+  id: string;
+  title: string;
+  content: string;
+}
```
Comment (Adona12) [inline Web/starter_project_barbies/pages/api/blogs/index.ts:7]: We already have this on the type folder

File: Web/starter_project_barbies/pages/blogs/[id].tsx [hunk 2]
```diff
@@ -1,79 +1,67 @@
...
+  if (result.isLoading) {
+    return <div>loading...</div>
+  }
+  
+  else if (result.isError) {
+    return <div>{result.error.toString()}</div>
   }
+  
+  else if (result.isSuccess) {
+    let blog = result.data
```
Comment (Adona12) [inline Web/starter_project_barbies/pages/blogs/[id].tsx:23]: How about destructuring this?

**Comment by YiseBoge** [comment]
i would have loved descriptions on this pr about what it does and what the assumptions are (it is especially important when working on the more complicated code bits)

**Comment by nawrazi** [comment]
> i would have loved descriptions on this pr about what it does and what the assumptions are (it is especially important when working on the more complicated code bits)
Done

File: Web/starter_project_barbies/pages/api/blogs/[id].ts
[the remark this answers is not in the document]
Comment (YiseBoge) [reply Web/starter_project_barbies/pages/api/blogs/[id].ts:1]: if you are indeed using a json server api, i believe you should define the base url as an env vairable and just load it from rtk query

Is there anything that would prevent that?

**Review by YiseBoge — CHANGES_REQUESTED** [review]
looks nice but i have left some changes to make

**Review by YiseBoge — CHANGES_REQUESTED** [review]
looks nice but i have left some changes to make

File: Web/starter_project_barbies/types/blog.ts [hunk 3]
```diff
@@ -16,5 +16,5 @@ export type Blog = {
     blogImage: string;
     readTime: number;
     shortDescription: string;
-    // we can add other properties here as needed
+    relatedBlogs: string[];
```
[the remark this answers is not in the document]
Comment (nawrazi) [reply Web/starter_project_barbies/types/blog.ts:19]: Yes, a list of string ID's.

**Comment by nawrazi** [comment]
@YiseBoge I have addressed all comments, please re-review.

File: Web/starter_project_barbies/pages/blogs/index.tsx [hunk 4]
```diff
@@ -1,58 +1,53 @@
...
-  return (
-    <div className='bg-white text-primary-text font-montserrat'>
-
-      {/* Title and search bar */}
-      <div className="flex justify-center pt-20">
-        <div className='grid grid-cols-3 mt-2 w-full'>
-          <div className="flex items-center">
-            <div className='ps-28 text-xl font-semibold'>
-              Blogs
+  const blogsResult = useGetBlogsQuery()
```
Comment (kkstarkk) [inline Web/starter_project_barbies/pages/blogs/index.tsx:8]: again use object destructuring

**Review by kkstarkk — CHANGES_REQUESTED** [review]
looks good overall and i have left some minor comments

**Review by kkstarkk — APPROVED** [review]

**Review by ruthgdb — APPROVED** [review]

**Pull request merged**